File: dp_labs_and_solutions/dp_labs_and_solutions/dp_java_labs_and_solutions/null_object/lab/UnaryMinus.java
// UnaryMinus.java

public class UnaryMinus extends Unary
{
	public UnaryMinus(Expression aOperand1)
	{
		super(aOperand1);
	}

	public void accept(ExpressionVisitor aExpressionVisitor)
	{
		aExpressionVisitor.visit(this);
	}
}

File: dp_labs_and_solutions/dp_labs_and_solutions/dp_cpp_labs_and_solutions/null_object/solution/unary_minus.cpp
// unary_minus.cpp

class expression;
#include "expression_visitor.h"
#include "unary.h"
#include "unary_minus.h"

unary_minus::unary_minus(expression * a_operand1)
: unary(a_operand1)
{
}

void unary_minus::accept(expression_visitor & a_expression_visitor)
{
	a_expression_visitor.visit(*this);
}
